app/assets/javascripts/discourse/app/components/user-menu/profile-tab-content.js

```diff
@@ -4,6 +4,16 @@ import { action } from "@ember/object";
 import showModal from "discourse/lib/show-modal";
 import DoNotDisturb from "discourse/lib/do-not-disturb";
 
+let _extraItems = [];
+
+export function addUserMenuProfileTabItem(item) {
+  _extraItems.push(item);
+}
+
+export function resetUserMenuProfileTabItems() {
+  _extraItems = [];
+}
```

Comment:
```suggestion
const EXTRA_ITEMS = [];

export function addUserMenuProfileTabItem(item) {
  EXTRA_ITEMS.push(item);
}

export function resetUserMenuProfileTabItems() {
  EXTRA_ITEMS.length = 0;
}
```

Since this is considered some sort of global, it is actually safer to mutate the existing array and clearing it by setting the length to `0`. Reassigning the variable to a new array can potentially lead to problems. See https://stackoverflow.com/a/1232046/2531506 for more details.

Reply:
FWIW I think we can use [`.clear()`](https://api.emberjs.com/ember/3.14/classes/MutableArray/methods/clear?anchor=clear) to remove all items from the existing array and avoid this "ugly" way of setting the length to 0.